### ISSUE TITLE
Add documentation and improved errors for QueryFn in MongoDbIO

### ISSUE THREAD

**Issue by damccorm** [apache/beam#21005]
Currently the MongoDbIO.Read implementation assumes that the submitted QueryFn is one of the predefined QueryFns in Beam, so either an AggregationQuery or FindQuery. The problem is this assumption is not documented anywhere.

Fixing this consists of two small changes:

1. Add documentation to the [withQueryFn](https://github.com/apache/beam/blob/release-2.31.0/sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java#L279) method specifying that the QueryFn must be one of the predefined classes in the MongoDbIO package.

2. In the [split method](https://github.com/apache/beam/blob/release-2.31.0/sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java#L435), the code checks if the QueryFn is a FindQuery, and if not assumes it's an AggregationQuery, meaning that users that input their own class will find it failing to cast to an AggregationQuery. Instead, there should be an error if the QueryFn is an unknown type, preferably instructing users to only use one of the predefined QueryFn types.

Imported from Jira [BEAM-12466](https://issues.apache.org/jira/browse/BEAM-12466). Original Jira may contain additional context.
Reported by: danoliveira.

**Comment by blazingbhavneek** [comment]
Hey there! 👋 I'm new to this repository and eager to contribute! 🌟 Could you kindly suggest some entry point or files to look into?

**Comment by damccorm** [comment]
Hey, saw you added this comment several places. I'd recommend focusing on a single issue at first (I answered the underlying question here - https://github.com/apache/beam/issues/20298#issuecomment-1547888993)

**Comment by Asmita0915** [comment]
Hey, I am willing to contribute for this issue. Don't see any assignee, Is this issue taken

**Comment by damccorm** [comment]
@Asmita0915 I don't think so, if you would like to take it you can comment `.take-issue` and a bot will auto-assign it your way

**Comment by Asmita0915** [comment]
.take-issue

**Comment by Asmita0915** [comment]
@damccorm taking a look might need your help if stuck. Thanks!

**Comment by Asmita0915** [comment]
@damccorm  i have dropped email to dev mailing list. Let me know if that looks fine.

**Comment by damccorm** [comment]
@johnjcasey could you help here?

**Comment by johnjcasey** [comment]
Yep, will do

**Comment by derrickaw** [comment]
.take-issue
